### ISSUE TITLE
Additional chat functionality

### ISSUE THREAD

**Issue by Pjb518** [UnicornHeartClub/Issues#25]
The in game chat needs a number of additional features:

- [ ] A way to clear the chat. My preference would be that the chat is cleared for an individual, and nothing is actually deleted. It's simply a way to clear the view.
- [ ] A way to delete individual items from the chat as a DM.
- [ ] A way to search the chat.
- [ ] The ability to send private messages in game.
- [ ] The ability to send messages using a language tag, which shows gibberish for characters who lack a  given language proficiency.

**Comment by Aquaricat** [comment]
I'll add
- [ ] The ability to export part, or all of the history as plain text and JSON.  
  - For the statistic nuts, the exports should record individual dice from batch rolls like Fireball.
- [ ] The ability for the DM to temporarily lock chat from receiving additional messages and/or rolls.
- [ ] The ability to enter a testing mode where macro output is displayed only to the person triggering the macro, and automatically deleted when the person exits test mode.
- [ ]  The ability to send the output of a macro to all players/dms, some players/dms, dm only, self only, and self+dm only.

@Pjb518 For the chat clearing - I'd argue that we actually need the ability to do both - permanent deletion, but also an archiving function.  Thoughts?

**Comment by Pjb518** [comment]
Regarding chat logs, I added a separate issue #26 for chat logs, which should DMs at least should be able to delete. This would be permanent deletion as you describe. I'd want to avoid users being able to delete messages because there are several ways people might use this to cheat, for example if the group rolls for stats in game, but doesn't do so through a session 0.

I do think that DMs should be able to clear or delete messages directly from the game though, so this would be an extension to the functionality requested in #26 : deletion of chat items in game as a means of deleting items from chat logs.

**Comment by Aquaricat** [comment]
> I'd want to avoid users being able to delete messages because there are several ways people might use this to cheat, for example if the group rolls for stats in game, but doesn't do so through a session 0

Definitely, some of this functionality is absolutely DM-only permission level.
